service/history/queues/queue_base.go

```diff
@@ -130,6 +136,10 @@ func newQueueBase(
 		metricsHandler,
 	)
 
+	monitor := newMonitor(category.Type(), &options.MonitorOptions)
+	mitigator, actionCh := newMitigator(monitor, logger, metricsHandler, options.MaxReaderCount)
+	monitor.registerMitigator(mitigator)
```

Comment:
there is circular dependency between mitigator and monitor. Any way we can break that?

Reply:
yeah... I can define an alertCh in the queueBase, and use it to first initialize monitor, then use alertCh and monitor to initialize mitigator. Then monitor doesn't need to know about mitigator, it only sends alerts to the alertCh. The mitigator interface can also be removed.

Reply:
Did some refactoring. Now monitor will send alert to an AlertCh. Mitigator will return an Action upon getting an alert instead of sending it to a separate channel. Queue basically get an alert from alertCh, call mitigator to get an action and then run the action. No more circular dependency as monitor no long knows about mitigator.

Reply:
Monitor is not used by mitigator in this PR, but it will be used in later PRs.